base/util/src/main/java/com/netscape/cmsutil/crypto/CryptoUtil.java

```diff
@@ -1611,6 +1611,41 @@ public static PKCS10 createCertificationRequest(String subjectName,
         return extn;
     }
 
+    /*
+     * addCTpoison adds the Certificate Transparency V1 poison extension
+     * to the Ceritificate Info
+     *
+     * @param certinfo X509CertInfo where the poison extension is to be added
+     *
+     * cfu
+     */
+    public static final String CT_POISON_OID = "1.3.6.1.4.1.11129.2.4.3";
+    public static final boolean CT_POISON_CRITICAL = true;
+    public static final byte CT_POISON_DATA[] =  new byte[] { 0x05, 0x00 };
+
+    public static void addCTpoison(X509CertInfo certinfo)
+                throws Exception {
+        String method = "CryptoUtil:addCTpoison: ";
+        ObjectIdentifier ct_poison_oid = new ObjectIdentifier(CT_POISON_OID);
+        Extension ct_poison_ext = null;
+        CertificateExtensions exts =  null;
+
+        exts = (CertificateExtensions)
+                certinfo.get(X509CertInfo.EXTENSIONS);
+        if (exts == null)
+            throw new Exception(method + " X509CertInfo.EXTENSIONS not found inf cetinfo");
```

Comment:
We should try throwing `org.mozilla.jss.netscape.security.x509.X509ExtensionException` rather than a generic exception